core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/ValidationApproachFactoryTest.java

```diff
@@ -9,30 +9,56 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+import java.util.Optional;
+
+import org.eclipse.rdf4j.common.transaction.TransactionSetting;
+import org.eclipse.rdf4j.common.transaction.TransactionSettingFactory;
 import org.eclipse.rdf4j.common.transaction.TransactionSettingRegistry;
-import org.eclipse.rdf4j.sail.shacl.ShaclSail.TransactionSettings.ValidationApproach;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
 
 /**
- * Unit tests for the {@link ValidationApproachFactory} class
+ * Tests that all {@link ShaclSail.TransactionSettings} are registered in the TransactionSettingRegistry through a
+ * TransactionSettingFactory
  *
- * @author Jeen Broekstra
+ * @author Jeen Broekstra, Håvard M. Ottestad
  *
  */
 public class ValidationApproachFactoryTest {
```

Comment:
Is this still the right name for this test class? I'm not sure I fully follow what's going on here but it looks like this is testing the functionality of the registry, rather than the `ValidationApproachFactory`.

Reply:
Yeah. I didn't update the name of the test class. I'll give it a new name.